### PR TITLE
Start fixing the easiest StyleCop issues

### PR DESCRIPTION
**Pull request by remi-stripe**
This PR explicitly ignores the most common StyleCop issues and fixes the less common ones to make sure it's the right direction.

cc @ob-stripe 
cc @stripe/api-libraries

### CODE REVIEW

File: src/Stripe.net/Stripe.net.csproj [hunk 1]
```diff
@@ -19,6 +19,7 @@
     <PackageTargetFallback Condition=" '$(TargetFramework)' == 'netstandard1.2' ">$(PackageTargetFallback);netcoreapp1.0</PackageTargetFallback>
     <NetStandardImplicitPackageVersion Condition=" '$(TargetFramework)' == 'netstandard1.2' ">1.6.1</NetStandardImplicitPackageVersion>
     <SignAssembly>True</SignAssembly>
+    <CodeAnalysisRuleSet>Settings.stylecop</CodeAnalysisRuleSet>
```
Comment (olivierbellone) [inline src/Stripe.net/Stripe.net.csproj]: I don't think you need this one.

Comment (remi-stripe) [reply src/Stripe.net/Stripe.net.csproj]: ah yes sorry forgot that I copy/pasted that everywhere to try to make it work :p

**Comment by remi-stripe** [comment]
Okay I pushed fixes for almost all the rules. There are a few left and I added comments as to what we need to agree on first.

File: src/Stripe.net/Infrastructure/Middleware/ParserPlugin/DateFilterPlugin.cs [hunk 2]
```diff
@@ -14,7 +14,7 @@ public bool Parse(ref string requestString, JsonPropertyAttribute attribute, Pro
                 return false;
             }
 
-            var filter = (StripeDateFilter) propertyValue;
+            var filter = (StripeDateFilter)propertyValue;
```
Comment (remi-stripe) [inline src/Stripe.net/Infrastructure/Middleware/ParserPlugin/DateFilterPlugin.cs]: While I implemented that one, I feel quite strongly that this missing space is not nice to look at and reason with.

**Comment by remi-stripe** [comment]
Okay almost all rules have been fixed. The ones left are a bit more complex or just painful (headers on all files) and left disabled for now. We also have 2 rules introducing breaking changes that we will fix on the next major release.

r? @ob-stripe

**Comment by brandur-stripe** [comment]
OMG Remi you're a BOSS. Amazing work here.

**Comment by remi-stripe** [comment]
@ob-stripe Okay I was able to fix the rule requiring `this` for local calls/variables after all with a few regexes.

I think this is as far as we can go without breaking changes so it's ready to be reviewed/merged.

**Comment by ob-stripe** [comment]
Amazing @remi-stripe!

Even with `?w=1`, it's kind of difficult to review this PR very closely so I'm largely relying on the test suite here, but lgtm.

**Pull request merged**